test/metabase/api/card_test.clj

```diff
@@ -1838,6 +1840,263 @@
+             :parameters [{:name "Category_name"
+                           :slug "category_name"
+                           :id   "_CATEGORY_NAME_"
+                           :type "category"}
+                          {:name "Category ID"
+                           :slug "category_id"
+                           :id   "_CATEGORY_ID_"
+                           :type "category"}
+                          {:name "Price"
+                           :slug "price"
+                           :id   "_PRICE_"
+                           :type "category"}
+                          {:name "id"
+                           :slug "id"
+                           :id   "_ID_"
+                           :type "category"}]
+             :parameter_mappings [{:parameter_id "_CATEGORY_NAME_"
+                                   :card_id      (:id card)
+                                   :target       [:template-tag {:id "c7fcf1fa"}]}
+                                  {:parameter_id "_CATEGORY_ID_"
+                                   :card_id      (:id card)
+                                   :target       [:template-tag {:id "a3cd3f3b"}]}
+                                  {:parameter_id "_PRICE_"
+                                   :card_id      (:id card)
+                                   :target       [:template-tag {:id "b879a8cd"}]}
+                                  {:parameter_id "_PRICE_NUMBER_"
+                                   :card_id      (:id card)
+                                   :target       [:template-tag {:id "e2ca38c9"}]}]})]
+      (db/update! Card (:id card)
+                  (merge card-defaults card-values)))
+    (f {:card       card
+        :param-keys {:category-name "_CATEGORY_NAME_"
+                     :category-id   "_CATEGORY_ID_"
+                     :price         "_PRICE_"
+                     :id            "_ID_"}})))
+
+
+(defmacro ^:private with-chain-filter-fixtures
+  "Create a query and its parameters."
+  {:style/indent 2}
+  [query-type [binding card-values] & body]
+  `(do-with-chain-filter-fixtures ~query-type ~card-values (fn [~binding] ~@body)))
+
+(defn add-query-params
+  "Encode query-params to an url."
+  [url query-params]
+  (let [query-params-str (str/join "&" (for [[k v] (partition 2 query-params)]
+                                         (codec/form-encode {k v})))]
+    (cond-> url
+      (seq query-params-str) (str "?" query-params-str))))
+
+(defn- chain-filter-values-url [card-or-id param-key & query-params]
+  (add-query-params (format "card/%d/params/%s/values" (u/the-id card-or-id) (name param-key))
+                    query-params))
+
+(defn- chain-filter-search-url [card-or-id param-key query & query-params]
+  (add-query-params (str (format "card/%d/params/%s/search/" (u/the-id card-or-id) (name param-key))
+                         query)
+                    query-params))
+
+(deftest chain-filter-test
```

Comment:
the list of tests are similar to tests on [dashboard](https://github.com/metabase/metabase/blob/98b81bf61aee0c2bc10e7def88bc84a04626d0d0/test/metabase/api/dashboard_test.clj#L1374)

The differences on card is that the parameter should work with both Notebook queries, and native queries